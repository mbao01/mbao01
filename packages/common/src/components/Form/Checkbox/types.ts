import { type VariantProps } from "../../../libs";
import { getCheckboxClasses } from "./constants";

export type CheckboxProps = Omit<
  React.HTMLProps<HTMLInputElement>,
  "size" | "ref" | "type"
> &
  VariantProps<typeof getCheckboxClasses>;
