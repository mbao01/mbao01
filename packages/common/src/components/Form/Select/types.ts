import { ReactNode } from "react";
import * as SelectPrimitive from "@radix-ui/react-select";
import { type VariantProps } from "../../../libs";
import {
  getLabelForSelectClasses,
  getSelectContentClasses,
  getSelectTriggerClasses,
  getSelectValueClasses,
} from "./constants";

export type SelectTriggerProps = React.ComponentPropsWithoutRef<typeof SelectPrimitive.Trigger> &
  VariantProps<typeof getSelectTriggerClasses> & {
    label?: ReactNode;
    labelPosition?: "start" | "end" | "floating";
  };

export type SelectItemProps = React.ComponentPropsWithoutRef<typeof SelectPrimitive.Item> &
  VariantProps<typeof getSelectTriggerClasses>;

export type SelectContentProps = React.ComponentPropsWithoutRef<typeof SelectPrimitive.Content> &
  VariantProps<typeof getSelectContentClasses>;

export type SelectValueProps = React.ComponentPropsWithoutRef<typeof SelectPrimitive.Value> &
  VariantProps<typeof getSelectValueClasses>;

export type LabelForSelectProps = VariantProps<typeof getLabelForSelectClasses> & {
  children: ReactNode;
  className?: string;
};
