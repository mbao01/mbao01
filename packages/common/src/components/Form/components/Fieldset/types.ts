import { type VariantProps } from "../../../../libs";
import { getFieldsetClasses, getFieldsetLabelClasses, getFieldsetLegendClasses } from "./constants";

export type FieldsetProps = React.FieldsetHTMLAttributes<HTMLFieldSetElement> &
  VariantProps<typeof getFieldsetClasses>;

export type As = "div" | "span" | "label" | "p";

type AllowedElements = Pick<React.JSX.IntrinsicElements, As>;

export type FieldsetLabelProps<T extends As> = (T extends "label"
  ? React.LabelHTMLAttributes<HTMLLabelElement>
  : AllowedElements[T]) &
  VariantProps<typeof getFieldsetLabelClasses> & {
    as?: T;
  };

export type FieldsetLegendProps = React.HTMLAttributes<HTMLLegendElement> &
  VariantProps<typeof getFieldsetLegendClasses>;
