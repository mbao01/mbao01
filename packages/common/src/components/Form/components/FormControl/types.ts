import { type VariantProps } from "../../../../libs";
import { getFormControlClasses } from "./constants";

export type As = "div" | "span" | "label";

type AllowedElements = Pick<React.JSX.IntrinsicElements, As>;

export type FormControlProps<T extends As> = (T extends "label"
  ? React.LabelHTMLAttributes<HTMLLabelElement>
  : React.HTMLAttributes<HTMLElement & AllowedElements[T]>) &
  VariantProps<typeof getFormControlClasses> & {
    as?: T;
  };
