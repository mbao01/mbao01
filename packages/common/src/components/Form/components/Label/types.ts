import * as LabelPrimitive from "@radix-ui/react-label";
import { type VariantProps } from "../../../../libs";
import { Text } from "../../../Text";
import { getLabelClasses } from "./constants";

export type LabelProps = React.ComponentPropsWithoutRef<typeof LabelPrimitive.Root> &
  VariantProps<typeof getLabelClasses>;

export type LabelTextProps = Omit<React.ComponentProps<typeof Text>, "as">;
