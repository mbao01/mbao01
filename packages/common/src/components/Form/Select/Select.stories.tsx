import type { Meta, StoryContext, StoryFn, StoryObj } from "@storybook/react-vite";
import { ReactNode } from "react";
import { Select } from "./Select";

type ArgsType = Partial<{
  size: "xs" | "sm" | "md" | "lg";
  wide: boolean;
  outline: boolean;
  variant:
    | "default"
    | "primary"
    | "secondary"
    | "accent"
    | "success"
    | "warning"
    | "info"
    | "error"
    | "ghost";
  label: ReactNode;
  labelPosition: "start" | "end" | "floating";
  placeholder: string;
  className: string;
}>;

const withSelect = (_: StoryFn, context: StoryContext<ArgsType>) => {
  const { size, wide, variant, outline, label, labelPosition, placeholder, className } =
    context.args;

  return (
    <Select>
      <Select.Trigger
        size={size}
        wide={wide}
        variant={variant}
        outline={outline}
        label={label}
        labelPosition={labelPosition}
        className={className}
      >
        <Select.Value placeholder={placeholder} />
      </Select.Trigger>
      <Select.Content>
        <Select.Item value="light" variant={variant}>
          Light
        </Select.Item>
        <Select.Item value="dark" variant={variant}>
          Dark
        </Select.Item>
        <Select.Item value="system" variant={variant}>
          System
        </Select.Item>
      </Select.Content>
    </Select>
  );
};

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
  title: "Components/Form/Select",
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
    layout: "centered",
    a11y: {
      config: {
        rules: [
          {
            // @fixme
            id: "button-name",
            reviewOnFail: true,
          },
        ],
      },
    },
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
  tags: ["autodocs"],
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  args: {
    wide: false,
    outline: false,
    size: undefined,
    variant: undefined,
  },
  argTypes: {
    wide: { control: "boolean" },
    outline: { control: "boolean" },
    size: {
      control: "select",
      options: ["xs", "sm", "md", "lg"],
    },
    variant: {
      control: "select",
      options: [
        "default",
        "primary",
        "secondary",
        "accent",
        "success",
        "warning",
        "info",
        "error",
        "ghost",
      ],
    },
  },
  decorators: [withSelect],
} satisfies Meta<ArgsType>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default: Story = {
  args: {
    placeholder: "Theme",
  },
};

export const PrimarySelect: Story = {
  args: {
    variant: "primary",
    placeholder: "Theme",
  },
};

export const OutlineSelect: Story = {
  args: {
    outline: true,
    placeholder: "Theme",
  },
};

export const TinySelect: Story = {
  args: {
    outline: true,
    size: "xs",
    placeholder: "Theme",
  },
};

export const FullWidthSelect: Story = {
  args: {
    outline: true,
    wide: true,
    placeholder: "Theme",
  },
};

export const LabelForSelect: Story = {
  args: {
    label: "Type",
    wide: true,
    outline: true,
    placeholder: "Theme",
  },
};

export const SuffixLabelForSelect: Story = {
  args: {
    label: "Be sure!",
    labelPosition: "end",
    wide: true,
    outline: true,
    placeholder: "Theme",
  },
};

export const FloatingLabelForSelect: Story = {
  args: {
    label: "Your Email",
    labelPosition: "floating",
    wide: true,
    outline: true,
    className: "min-w-xs",
  },
};
