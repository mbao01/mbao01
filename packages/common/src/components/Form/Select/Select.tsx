"use client";

import * as React from "react";
import * as SelectPrimitive from "@radix-ui/react-select";
import { CheckIcon, ChevronDownIcon, ChevronUpIcon } from "lucide-react";
import type { SelectContentProps, SelectItemProps, SelectTriggerProps } from "./types";
import { cn } from "../../../utilities";
import {
  getSelectContentClasses,
  getSelectItemClasses,
  getSelectLabelClasses,
  getSelectSeparatorClasses,
  getSelectTriggerClasses,
  getSelectViewportClasses,
} from "./constants";

const Select = (props: React.ComponentPropsWithoutRef<typeof SelectPrimitive.Root>) => (
  <SelectPrimitive.Root {...props} />
);

const SelectTrigger = React.forwardRef<
  React.ElementRef<typeof SelectPrimitive.Trigger>,
  SelectTriggerProps
>(({ className, children, size, variant, outline, ...props }, ref) => (
  <SelectPrimitive.Trigger
    ref={ref}
    className={cn(getSelectTriggerClasses({ size, variant, outline }), className)}
    {...props}
  >
    {children}
  </SelectPrimitive.Trigger>
));
SelectTrigger.displayName = SelectPrimitive.Trigger.displayName;

const SelectScrollUpButton = React.forwardRef<
  React.ElementRef<typeof SelectPrimitive.ScrollUpButton>,
  React.ComponentPropsWithoutRef<typeof SelectPrimitive.ScrollUpButton>
>((props, ref) => (
  <SelectPrimitive.ScrollUpButton
    ref={ref}
    className={cn("flex cursor-default items-center justify-center py-1")}
    {...props}
  >
    <ChevronUpIcon />
  </SelectPrimitive.ScrollUpButton>
));
SelectScrollUpButton.displayName = SelectPrimitive.ScrollUpButton.displayName;

const SelectScrollDownButton = React.forwardRef<
  React.ElementRef<typeof SelectPrimitive.ScrollDownButton>,
  React.ComponentPropsWithoutRef<typeof SelectPrimitive.ScrollDownButton>
>((props, ref) => (
  <SelectPrimitive.ScrollDownButton
    ref={ref}
    className={cn("flex cursor-default items-center justify-center py-1")}
    {...props}
  >
    <ChevronDownIcon />
  </SelectPrimitive.ScrollDownButton>
));
SelectScrollDownButton.displayName = SelectPrimitive.ScrollDownButton.displayName;

const SelectContent = React.forwardRef<
  React.ElementRef<typeof SelectPrimitive.Content>,
  SelectContentProps
>(({ className, children, position = "popper", ...props }, ref) => (
  <SelectPrimitive.Portal>
    <SelectPrimitive.Content
      ref={ref}
      className={cn(getSelectContentClasses({ position }), className)}
      position={position}
      {...props}
    >
      <SelectScrollUpButton />
      <SelectPrimitive.Viewport className={cn(getSelectViewportClasses({ position }))}>
        {children}
      </SelectPrimitive.Viewport>
      <SelectScrollDownButton />
    </SelectPrimitive.Content>
  </SelectPrimitive.Portal>
));
SelectContent.displayName = SelectPrimitive.Content.displayName;

const SelectLabel = React.forwardRef<
  React.ElementRef<typeof SelectPrimitive.Label>,
  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Label>
>((props, ref) => (
  <SelectPrimitive.Label ref={ref} className={cn(getSelectLabelClasses())} {...props} />
));
SelectLabel.displayName = SelectPrimitive.Label.displayName;

const SelectItem = React.forwardRef<React.ElementRef<typeof SelectPrimitive.Item>, SelectItemProps>(
  ({ className, children, variant, ...props }, ref) => (
    <SelectPrimitive.Item
      ref={ref}
      className={cn(getSelectItemClasses({ variant }), className)}
      {...props}
    >
      <span className="absolute right-2 flex h-3.5 w-3.5 items-center justify-center">
        <SelectPrimitive.ItemIndicator>
          <CheckIcon className="h-4 w-4" />
        </SelectPrimitive.ItemIndicator>
      </span>
      <SelectPrimitive.ItemText>{children}</SelectPrimitive.ItemText>
    </SelectPrimitive.Item>
  )
);
SelectItem.displayName = SelectPrimitive.Item.displayName;

const SelectSeparator = React.forwardRef<
  React.ElementRef<typeof SelectPrimitive.Separator>,
  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Separator>
>((props, ref) => (
  <SelectPrimitive.Separator ref={ref} className={cn(getSelectSeparatorClasses())} {...props} />
));
SelectSeparator.displayName = SelectPrimitive.Separator.displayName;

Select.Group = SelectPrimitive.Group;
Select.Value = SelectPrimitive.Value;
Select.Trigger = SelectTrigger;
Select.Content = SelectContent;
Select.Label = SelectLabel;
Select.Item = SelectItem;
Select.Icon = SelectPrimitive.Icon;
Select.ItemText = SelectPrimitive.ItemText;
Select.ItemIndicator = SelectPrimitive.ItemIndicator;
Select.Separator = SelectSeparator;
Select.ScrollUpButton = SelectScrollUpButton;
Select.ScrollDownButton = SelectScrollDownButton;

export { Select };
