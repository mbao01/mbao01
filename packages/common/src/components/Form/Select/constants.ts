import { cva } from "../../../libs";

export const getSelectTriggerClasses = cva(
  "select flex items-center justify-between rounded-md text-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1",
  {
    variants: {
      variant: {
        primary: "select-primary",
        secondary: "select-secondary",
        accent: "select-accent",
        success: "select-success",
        warning: "select-warning",
        info: "select-info",
        error: "select-error",
      },
      outline: {
        true: "select-bordered border border-input",
      },
      wide: {
        true: "w-full",
      },
      size: {
        xs: "select-xs",
        sm: "select-sm",
        md: "select-md",
        lg: "select-lg",
      },
    },
    compoundVariants: [
      {
        size: undefined,
        className: "min-h-fit h-10",
      },
    ],
  }
);

export const getSelectItemClasses = cva(
  "relative flex w-full cursor-pointer select-none items-center focus:bg-neutral focus:text-neutral-content rounded-sm py-1.5 pl-2 pr-8 text-sm outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50",
  {
    variants: {
      variant: {
        primary: "focus:bg-primary focus:text-primary-content",
        secondary: "focus:bg-secondary focus:text-secondary-content",
        accent: "focus:bg-accent focus:text-accent-content",
        success: "focus:bg-success focus:text-success-content",
        warning: "focus:bg-warning focus:text-warning-content",
        info: "focus:bg-info focus:text-info-content",
        error: "focus:bg-error focus:text-error-content",
      },
    },
  }
);

export const getSelectLabelClasses = cva("px-2 py-1.5 text-sm font-semibold");

export const getSelectSeparatorClasses = cva("-mx-1 my-1 h-px bg-muted");

export const getSelectContentClasses = cva(
  "relative z-50 max-h-96 min-w-[8rem] overflow-hidden rounded-md border bg-base-100 shadow-md data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2",
  {
    variants: {
      position: {
        popper:
          "data-[side=bottom]:translate-y-1 data-[side=left]:-translate-x-1 data-[side=right]:translate-x-1 data-[side=top]:-translate-y-1",
        "item-aligned": "",
      },
    },
  }
);

export const getSelectViewportClasses = cva("p-1", {
  variants: {
    position: {
      popper:
        "h-[var(--radix-select-trigger-height)] w-full min-w-[var(--radix-select-trigger-width)]",
      "item-aligned": "",
    },
  },
});
