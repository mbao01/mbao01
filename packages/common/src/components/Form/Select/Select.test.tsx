import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { Select } from "./Select";
import { type SelectTriggerProps } from "./types";

describe("Select", () => {
  const renderSelect = (
    placeholder = "Theme",
    { size, variant, outline }: Partial<SelectTriggerProps> = {}
  ) => {
    return render(
      <Select>
        <Select.Trigger size={size} variant={variant} outline={outline}>
          <Select.Value placeholder={placeholder} />
        </Select.Trigger>
        <Select.Content>
          <Select.Item value="light" variant={variant}>
            Light
          </Select.Item>
          <Select.Item value="dark" variant={variant}>
            Dark
          </Select.Item>
          <Select.Item value="system" variant={variant}>
            System
          </Select.Item>
        </Select.Content>
      </Select>
    );
  };

  it("renders with the correct initial state", () => {
    const { asFragment } = renderSelect();

    expect(screen.getByRole("combobox")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it.each(["primary", "secondary", "accent", "success", "warning", "info", "error"] as const)(
    "has %s variant",
    (variant) => {
      const placeholder = `${variant} select`;
      const { asFragment } = renderSelect(placeholder, { variant });

      expect(screen.getByRole("combobox")).toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    }
  );

  it.each([
    { size: "xs", name: "tiny" },
    { size: "sm", name: "small" },
    { size: "md", name: "base" },
    { size: "lg", name: "large" },
  ] as const)("has $name ($size)", ({ size, name }) => {
    const placeholder = `${name} select`;
    const { asFragment } = renderSelect(placeholder, { size });

    expect(screen.getByRole("combobox")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
