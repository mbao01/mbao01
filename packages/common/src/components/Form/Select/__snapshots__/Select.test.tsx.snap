// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Select > has 'base' ('md') 1`] = `
<DocumentFragment>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:r18:"
    aria-expanded="false"
    class="select flex items-center justify-between rounded-md text-sm text-left ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 select-md"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      base select
    </span>
  </button>
</DocumentFragment>
`;

exports[`Select > has 'large' ('lg') 1`] = `
<DocumentFragment>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:r1c:"
    aria-expanded="false"
    class="select flex items-center justify-between rounded-md text-sm text-left ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 select-lg"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      large select
    </span>
  </button>
</DocumentFragment>
`;

exports[`Select > has 'small' ('sm') 1`] = `
<DocumentFragment>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:r14:"
    aria-expanded="false"
    class="select flex items-center justify-between rounded-md text-sm text-left ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 select-sm"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      small select
    </span>
  </button>
</DocumentFragment>
`;

exports[`Select > has 'tiny' ('xs') 1`] = `
<DocumentFragment>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:r10:"
    aria-expanded="false"
    class="select flex items-center justify-between rounded-md text-sm text-left ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 select-xs"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      tiny select
    </span>
  </button>
</DocumentFragment>
`;

exports[`Select > has accent variant 1`] = `
<DocumentFragment>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:rc:"
    aria-expanded="false"
    class="select flex items-center justify-between rounded-md text-sm text-left ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 select-accent min-h-fit h-10"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      accent select
    </span>
  </button>
</DocumentFragment>
`;

exports[`Select > has error variant 1`] = `
<DocumentFragment>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:rs:"
    aria-expanded="false"
    class="select flex items-center justify-between rounded-md text-sm text-left ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 select-error min-h-fit h-10"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      error select
    </span>
  </button>
</DocumentFragment>
`;

exports[`Select > has info variant 1`] = `
<DocumentFragment>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:ro:"
    aria-expanded="false"
    class="select flex items-center justify-between rounded-md text-sm text-left ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 select-info min-h-fit h-10"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      info select
    </span>
  </button>
</DocumentFragment>
`;

exports[`Select > has primary variant 1`] = `
<DocumentFragment>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:r4:"
    aria-expanded="false"
    class="select flex items-center justify-between rounded-md text-sm text-left ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 select-primary min-h-fit h-10"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      primary select
    </span>
  </button>
</DocumentFragment>
`;

exports[`Select > has secondary variant 1`] = `
<DocumentFragment>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:r8:"
    aria-expanded="false"
    class="select flex items-center justify-between rounded-md text-sm text-left ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 select-secondary min-h-fit h-10"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      secondary select
    </span>
  </button>
</DocumentFragment>
`;

exports[`Select > has success variant 1`] = `
<DocumentFragment>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:rg:"
    aria-expanded="false"
    class="select flex items-center justify-between rounded-md text-sm text-left ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 select-success min-h-fit h-10"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      success select
    </span>
  </button>
</DocumentFragment>
`;

exports[`Select > has warning variant 1`] = `
<DocumentFragment>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:rk:"
    aria-expanded="false"
    class="select flex items-center justify-between rounded-md text-sm text-left ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 select-warning min-h-fit h-10"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      warning select
    </span>
  </button>
</DocumentFragment>
`;

exports[`Select > renders with the correct initial state 1`] = `
<DocumentFragment>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:r0:"
    aria-expanded="false"
    class="select flex items-center justify-between rounded-md text-sm text-left ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 min-h-fit h-10"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      Theme
    </span>
  </button>
</DocumentFragment>
`;
